mitodl/ocw-studio
# Sync missing captions, filepath issue, existing captions issue fixed

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes https://github.com/mitodl/ocw-studio/issues/1791

#### What's this PR do?
The PR adds an a flag `--delete-existing` 
Testing this PR has been very troublesome since once the captions and transcripts references are updated on Video resources, a celery syncs that data to the Github.
So to test the sync command, `sync_missing_captions`, reviewer had to run manual queries to remove those references before testing, resulting in increased `turnaround time`

Moreover, when testing the prior fix, https://github.com/mitodl/ocw-studio/pull/1772, we noticed the incorrect paths being formed which were not clear when testing locally. That has also been addressed in this PR.

#### How should this be manually tested?
- Make sure you have latest database dump locally
   - This is to ensure you have courses with missing captions/transcripts locally
- Make sure to add following .env variables: 
   - **Threeplay variables** 
      - `THREEPLAY_API_KEY`,  `THREEPLAY_CALLBACK_KEY`,  `THREEPLAY_PROJECT_ID`
   - **AWS variables**   
     - `AWS_STORAGE_BUCKET_NAME`,  `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`
-  Checkout to this branch
- **Running the command to:**
  - **Test a list of courses, also deleting existing captions**
    - `docker-compose run web ./manage.py sync_missing_captions --filter <website_short_id> --delete-existing`
    - This will clear `video_captions_url` and `video_transcript_file` and delete existing caption and transcript objects
  - **Test all courses without deleting previous captions**
     - Run command `docker-compose run web  ./manage.py sync_missing_captions`
  - **Test single course**
     - Run command `docker-compose run web  ./manage.py sync_missing_captions --filter <course-short_id>`
   - Add comma separated course names to be parsed
- Head to `http://localhost:9001` (Local Minio)
- Go to `ol-ocw-studio-app/courses/<course-name>`
- You should see new captions (\*_captions.webvtt) and transcripts (\*_transcript.pdf) file with some `Size` value (previously empty **-**)
